# Keyboard shortcuts

**Preciona H**

## Importar librerías

In [1]:
import pandas as pd
import numpy as np

## Leer Dataset

In [22]:
#Ruta donde se va a leer el dataset
path= 'Dataset/demanda-de-electricidad-datos-mensuales.csv'
#Se le asigna a una variable de python el dataset por medio de la funcion de Pandas "pd.read_csv()"
dataset= pd.read_csv(path)

### Primera exploración del dataset con pandas.
###### .head()
    la funcion head te muestra las 5 primeras filas de tu dataset.

In [23]:
dataset.head()

indice_tiempo  demanda_total  demanda_residencial  comercio_e_industria  \
0    2001-01-01        6866.26                  NaN                   NaN   
1    2001-02-01        6473.10                  NaN                   NaN   
2    2001-03-01        6952.30                  NaN                   NaN   
3    2001-04-01        6183.35                  NaN                   NaN   
4    2001-05-01        6668.94                  NaN                   NaN   

   grandes_usuarios  temperatura_promedio  potencia_maxima  ede_tucuman  \
0               NaN                25.432          13501.0          NaN   
1               NaN                25.596          14061.0          NaN   
2               NaN                22.852          13780.0          NaN   
3               NaN                18.317          12866.0          NaN   
4               NaN                14.306          12968.0          NaN   

   edelap_sa  edenor_distribuidor       ...         energia_de_catamarca_sa  \
0        NaN                  NaN       ...                             NaN   
1        NaN                  NaN       ...                             NaN   
2        NaN                  NaN       ...                             NaN   
3        NaN                  NaN       ...                             NaN   
4        NaN                  NaN       ...                             NaN   

   energia_de_entre_rios_sa  energia_de_mendoza_sa  \
0                       NaN                    NaN   
1                       NaN                    NaN   
2                       NaN                    NaN   
3                       NaN                    NaN   
4                       NaN                    NaN   

   energia_san_juan_sa_exedessa  epec_distribuidor  epen_distribuidor  \
0                           NaN                NaN                NaN   
1                           NaN                NaN                NaN   
2                           NaN                NaN                NaN   
3                           NaN                NaN                NaN   
4                           NaN                NaN                NaN   

   epesf_distribuidor  recursos_y_energia_formosa_sa  secheep  spse_santa_cruz  
0                 NaN                            NaN      NaN              NaN  
1                 NaN                            NaN      NaN              NaN  
2                 NaN                            NaN      NaN              NaN  
3                 NaN                            NaN      NaN              NaN  
4                 NaN                            NaN      NaN              NaN  

[5 rows x 30 columns]

## Columnas del Dataset

In [9]:
dataset.columns

Index(['indice_tiempo', 'demanda_total', 'demanda_residencial',
       'comercio_e_industria', 'grandes_usuarios', 'temperatura_promedio',
       'potencia_maxima', 'ede_tucuman', 'edelap_sa', 'edenor_distribuidor',
       'edesal_distribuidor', 'edestesa_emp_dist__este', 'edesur_distribuidor',
       'emp_de_energia_de_rio_negro_sa', 'emp_dist_energ_atlantica',
       'emp_de_energia_de_la_rioja_sa', 'emp_dist_energia_de_salta',
       'emp_electric_misiones_sa', 'empresa_dis_estero_sa',
       'empresa_jujenia_de_energia_sa', 'energia_de_catamarca_sa',
       'energia_de_entre_rios_sa', 'energia_de_mendoza_sa',
       'energia_san_juan_sa_exedessa', 'epec_distribuidor',
       'epen_distribuidor', 'epesf_distribuidor',
       'recursos_y_energia_formosa_sa', 'secheep', 'spse_santa_cruz'],
      dtype='object')

##### ¿Cuantas columnas tiene el data set?

In [11]:
len(dataset.columns)

30

##### ¿Cuales son las dimensiones del Dataset?

In [12]:
dataset.shape

(213, 30)

   Tenemos **213 Filas** y **30 Columnas**

## Tipos de columnas
#### .dtypes 
    Es un atributo del DataFrame que te da los tipos de datos por cada Columna

In [10]:
dataset.dtypes

indice_tiempo                      object
demanda_total                     float64
demanda_residencial               float64
comercio_e_industria              float64
grandes_usuarios                  float64
temperatura_promedio              float64
potencia_maxima                   float64
ede_tucuman                       float64
edelap_sa                         float64
edenor_distribuidor               float64
edesal_distribuidor               float64
edestesa_emp_dist__este           float64
edesur_distribuidor               float64
emp_de_energia_de_rio_negro_sa    float64
emp_dist_energ_atlantica          float64
emp_de_energia_de_la_rioja_sa     float64
emp_dist_energia_de_salta         float64
emp_electric_misiones_sa          float64
empresa_dis_estero_sa             float64
empresa_jujenia_de_energia_sa     float64
energia_de_catamarca_sa           float64
energia_de_entre_rios_sa          float64
energia_de_mendoza_sa             float64
energia_san_juan_sa_exedessa      

#### Que tipos de datos tiene Pandas
* **object**: Texto
* **float**: Numerico con decimales
* **int**: Numerico entero
* **bool**: True o False, Valores binarios.
* **category**: Valores categoricos
* **datetime ó timedelta**: Son Datos con fechas o tiempos

## Aplicamos slicing solo de los datos generales y de la compañía Epec

In [19]:
dataset_epec = dataset[['indice_tiempo','demanda_residencial','comercio_e_industria','grandes_usuarios','temperatura_promedio','epec_distribuidor']]
dataset_epec

indice_tiempo  demanda_residencial  comercio_e_industria  \
0      2001-01-01                  NaN                   NaN   
1      2001-02-01                  NaN                   NaN   
2      2001-03-01                  NaN                   NaN   
3      2001-04-01                  NaN                   NaN   
4      2001-05-01                  NaN                   NaN   
5      2001-06-01                  NaN                   NaN   
6      2001-07-01                  NaN                   NaN   
7      2001-08-01                  NaN                   NaN   
8      2001-09-01                  NaN                   NaN   
9      2001-10-01                  NaN                   NaN   
10     2001-11-01                  NaN                   NaN   
11     2001-12-01                  NaN                   NaN   
12     2002-01-01                  NaN                   NaN   
13     2002-02-01                  NaN                   NaN   
14     2002-03-01                  NaN                   NaN   
15     2002-04-01                  NaN                   NaN   
16     2002-05-01                  NaN                   NaN   
17     2002-06-01                  NaN                   NaN   
18     2002-07-01                  NaN                   NaN   
19     2002-08-01                  NaN                   NaN   
20     2002-09-01                  NaN                   NaN   
21     2002-10-01                  NaN                   NaN   
22     2002-11-01                  NaN                   NaN   
23     2002-12-01                  NaN                   NaN   
24     2003-01-01                  NaN                   NaN   
25     2003-02-01                  NaN                   NaN   
26     2003-03-01                  NaN                   NaN   
27     2003-04-01                  NaN                   NaN   
28     2003-05-01                  NaN                   NaN   
29     2003-06-01                  NaN                   NaN   
..            ...                  ...                   ...   
183    2016-04-01          4104.173856           4148.827000   
184    2016-05-01          4941.165168           4359.846460   
185    2016-06-01          5634.190447           4282.756603   
186    2016-07-01          5614.534651           4366.110724   
187    2016-08-01          4663.796035           4249.645917   
188    2016-09-01          4351.181774           4063.899360   
189    2016-10-01          3953.447552           3985.591323   
190    2016-11-01          3970.327674           4142.212910   
191    2016-12-01          5148.817670           4687.036718   
192    2017-01-01          5681.816626           4849.049020   
193    2017-02-01          4618.226100           4551.680399   
194    2017-03-01          4470.715122           4514.516433   
195    2017-04-01          3754.330119           3986.428466   
196    2017-05-01          4524.024791           4137.188624   
197    2017-06-01          5180.585335           4149.745327   
198    2017-07-01          5460.615925           4259.809832   
199    2017-08-01          4767.354968           4209.895737   
200    2017-09-01          4217.808567           4048.775981   
201    2017-10-01          3979.758234           4117.571079   
202    2017-11-01          4035.416488           4174.310998   
203    2017-12-01          5216.357135           4704.238326   
204    2018-01-01          5468.423592           4781.759728   
205    2018-02-01          4963.781739           4416.421143   
206    2018-03-01          4492.917162           4579.324971   
207    2018-04-01          4084.109752           4269.204833   
208    2018-05-01          4276.829777           4236.631730   
209    2018-06-01          5569.775499           4332.322478   
210    2018-07-01          6167.113772           4421.727938   
211    2018-08-01          5359.054186           4301.646845   
212    2018-09-01          3995.352693           3815.260592   

     grandes_usuarios  temperatura_

### Podemos buscar solo los datos que sean del 2018
###### Con la funcion de Pandas .loc

In [21]:
dataset_epec.loc[dataset_epec.indice_tiempo >= '2018']

indice_tiempo  demanda_residencial  comercio_e_industria  \
204    2018-01-01          5468.423592           4781.759728   
205    2018-02-01          4963.781739           4416.421143   
206    2018-03-01          4492.917162           4579.324971   
207    2018-04-01          4084.109752           4269.204833   
208    2018-05-01          4276.829777           4236.631730   
209    2018-06-01          5569.775499           4332.322478   
210    2018-07-01          6167.113772           4421.727938   
211    2018-08-01          5359.054186           4301.646845   
212    2018-09-01          3995.352693           3815.260592   

     grandes_usuarios  temperatura_promedio  epec_distribuidor  
204            2100.0                  26.1         832.616497  
205            2024.0                  25.2         778.396217  
206            2175.0                  22.4         775.204147  
207            2140.0                  22.1         715.794615  
208            2152.0                  16.6         732.903479  
209            2022.0                  11.3         848.118390  
210            2017.0                  10.8         915.482586  
211            2049.0                  12.4         818.226530  
212            1963.0                  17.5                NaN

## Clase que viene Empezaremos de lleno con Pandas